src/pathfind/teleport.cpp

```diff
@@ -218,53 +218,39 @@ teleport_map::teleport_map(
 		std::string teleport_id = group.get_teleport_id();
 		std::set<map_location>::iterator source_it = locations.first.begin();
 		for (; source_it != locations.first.end(); ++source_it ) {
-			if(teleport_map_.count(*source_it) == 0) {
-				std::set<std::string> id_set;
-				id_set.insert(teleport_id);
-				teleport_map_.emplace(*source_it, id_set);
+			auto map_it = teleport_map_.find(*source_it);
+
+			if(map_it == teleport_map_.end()) {
+				std::set<map_location> map_locations;
+				map_locations.insert(locations.second.begin(), locations.second.end());
```

Comment:
Something seems a little off about the change here (and a couple lines down). What happened to the `teleport_id` and why is it no longer being used here? I'm not sure what the implications of that will be, but I suspect it to be incorrect.

Reply:
It was used as a key to link the maps that are now merged therefore it is no longer needed. In the latest commit I removed it.